core/adminserver/admin_server.go

```diff
@@ -155,13 +155,13 @@ func (s *Server) fetchDomain(ctx context.Context, d *domain.Domain) (*pb.Domain,
 	}, nil
 }
 
-// GetDomain retrieves the domain info for a given domain.
-func (s *Server) GetDomain(ctx context.Context, in *pb.GetDomainRequest) (*pb.Domain, error) {
-	domain, err := s.domains.Read(ctx, in.GetDomainId(), in.GetShowDeleted())
+// GetDirectory retrieves the directory info for a given directory.
+func (s *Server) GetDirectory(ctx context.Context, in *pb.GetDirectoryRequest) (*pb.Directory, error) {
+	directory, err := s.directories.Read(ctx, in.GetDirectoryId(), in.GetShowDeleted())
 	if err != nil {
 		return nil, err
 	}
-	info, err := s.fetchDomain(ctx, domain)
+	info, err := s.fetchDirectory(ctx, directory)
```

Comment:
why don't we just return the result directly?

Reply:
we could